driver-scala/src/main/scala/org/mongodb/scala/DistinctObservable.scala

```diff
@@ -111,6 +111,30 @@ case class DistinctObservable[TResult](private val wrapped: DistinctPublisher[TR
     this
   }
 
+  /**
+   * Sets the hint for this operation. A null value means no hint is set.
+   *
+   * @param hint the hint
+   * @return this
+   * @since 5.3
+   */
+  def hint(hint: Bson): DistinctObservable[TResult] = {
+    wrapped.hint(hint)
+    this
+  }
+
+  /**
+   * Sets the hint for this operation. A null value means no hint is set.
+   *
+   * @param hint the name of the index which should be used for the operation
+   * @return this
+   * @since 5.3
```

Comment:
again, we'd better add the following key detail in line with `FindObservable`:
```
   * @note if [[hint]] is set that will be used instead of any hint string.
```